### ISSUE TITLE
[2.5.1/2.6] Carb absorption not done properly for SMB + Average Sensitivity

### ISSUE THREAD

**Issue by diascout** [MilosKozak/AndroidAPS#2344]
Time of event: Sun 5, 02:40
AndroidAPS 2.5.1 (2.6 might be impacted, too)
Build: fa1ffc9e6-2019.11.03-22:21
Flavor: fullrelease
Nightscout-Version: 0.12.5

I am testing SMBs but in combination with the Average Sensitivity instead of Oref1. The reason is that for my child, Oref1 doesn't act quickly enough to adjust the sensitivity.

You can see the challenge in the attached Screenshot, I included the logfiles, too. It happened that when I entered 4g carbs, only 2 got absorbed by AAPS, 2 got stuck. min_5m_carbimpact = 8. There is another challenge that in combination of Average Sensitivity + SMB, there is not data entry field to enter the min_5m_carbimpact. You have to select first either SMB+Oref1 to enter it in Oref1 or you select AMA+Average Sensitivity to enter the value in AMA. Afterwards, you can revert back to SMB + Average Sensitivity. It may contribute to the observed concern ?

![IMG-20200105-WA0000](https://user-images.githubusercontent.com/44416182/72064619-8bed4c80-32dc-11ea-8e5b-b65d60df6028.jpg)

[AndroidAPS_LOG_1578188390954.log.zip](https://github.com/MilosKozak/AndroidAPS/files/4040169/AndroidAPS_LOG_1578188390954.log.zip)

**Comment by MilosKozak** [comment]
for weighted average and aaps min5impact is calculated from max carbs absorption time.
that's why you not find it in preferences
check wiki

**Comment by diascout** [comment]
True, I remember it now! Nevertheless, in this specific example, min5impact seems to be too low leading to a slow carb absorption. Is there any possibility for me to check this in the log file what the value actually was?
Especially for SMBs, most of the articles about it state that min5impact should be by default 8 to avoid that Carbs linger around. I wonder whether in the combination of Weigthed Sensitivity + SMBs, the code should have for safety reasons some hardcoded threshold included to establish a lower limit for min5impact. When I logged this issue, I switched back to Oref1 and actually saw that the 4 carbs would have been absorbed much earlier.
